### DISCUSSION TITLE
ClamAV broken when updating to nextcloud 28 from 27

### DISCUSSION THREAD

**Discussion by Sabering1** [nextcloud/all-in-one#4477] [Questions] [answered]
### Steps to reproduce
1. Update to nextcloud 28 from 27
2. ClamAV will not be found and will halt the NextCloud main
3. After a while NextCloud will start but ClamAV will not

### Expected behavior 
All containers start with no issue

### Actual behavior 
ClamAV is broken

### Host OS 
Debian Gnu/Linux 12

#### Nextcloud AIO version 
8
#### Current channel 
Latest
#### Other valuable info 

Failed to start clamd
Starting Freshclamd
Starting ClamAV

or clamd not found yet, retrying (0/90) ...ClamAV update process started at Fri Mar 29 13:57:20 2024
daily.cld database is up-to-date (version: 27229, sigs: 2057112, f-level: 90, builder: raynman)
main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
bytecode.cld database is up-to-date (version: 335, sigs: 86, f-level: 90, builder: raynman)
LibClamAV Error: Can't load /var/lib/clamav/main.cvd: Can't verify database integrity
LibClamAV Error: cli_loaddbdir: error loading database /var/lib/clamav/main.cvd
ERROR: Fri Mar 29 13:57:27 2024 -> Can't verify database integrity

or clamd not found yet, retrying (1/90) ...
or clamd not found yet, retrying (2/90) ...
or clamd not found yet, retrying (3/90) ...
or clamd not found yet, retrying (4/90) ...
or clamd not found yet, retrying (5/90) ...
or clamd not found yet, retrying (6/90) ...
or clamd not found yet, retrying (7/90) ...
or clamd not found yet, retrying (8/90) ...
or clamd not found yet, retrying (9/90) ...
or clamd not found yet, retrying (10/90) ...
or clamd not found yet, retrying (11/90) ...
or clamd not found yet, retrying (12/90) ...
or clamd not found yet, retrying (13/90) ...
or clamd not found yet, retrying (14/90) ...
or clamd not found yet, retrying (15/90) ...
or clamd not found yet, retrying (16/90) ...
or clamd not found yet, retrying (17/90) ...
or clamd not found yet, retrying (18/90) ...
or clamd not found yet, retrying (19/90) ...
or clamd not found yet, retrying (20/90) ...
or clamd not found yet, retrying (21/90) ...
or clamd not found yet, retrying (22/90) ...
or clamd not found yet, retrying (23/90) ...
or clamd not found yet, retrying (24/90) ...
or clamd not found yet, retrying (25/90) ...
or clamd not found yet, retrying (26/90) ...
or clamd not found yet, retrying (27/90) ...
or clamd not found yet, retrying (28/90) ...
or clamd not found yet, retrying (29/90) ...
or clamd not found yet, retrying (30/90) ...
or clamd not found yet, retrying (31/90) ...
or clamd not found yet, retrying (32/90) ...
or clamd not found yet, retrying (33/90) ...
or clamd not found yet, retrying (34/90) ...
or clamd not found yet, retrying (35/90) ...
or clamd not found yet, retrying (36/90) ...
or clamd not found yet, retrying (37/90) ...
or clamd not found yet, retrying (38/90) ...
or clamd not found yet, retrying (39/90) ...
or clamd not found yet, retrying (40/90) ...
or clamd not found yet, retrying (41/90) ...
or clamd not found yet, retrying (42/90) ...
or clamd not found yet, retrying (43/90) ...
or clamd not found yet, retrying (44/90) ...
or clamd not found yet, retrying (45/90) ...
or clamd not found yet, retrying (46/90) ...
or clamd not found yet, retrying (47/90) ...
or clamd not found yet, retrying (48/90) ...
or clamd not found yet, retrying (49/90) ...
or clamd not found yet, retrying (50/90) ...
or clamd not found yet, retrying (51/90) ...
or clamd not found yet, retrying (52/90) ...
or clamd not found yet, retrying (53/90) ...
or clamd not found yet, retrying (54/90) ...
or clamd not found yet, retrying (55/90) ...
or clamd not found yet, retrying (56/90) ...
or clamd not found yet, retrying (57/90) ...
or clamd not found yet, retrying (58/90) ...
or clamd not found yet, retrying (59/90) ...
or clamd not found yet, retrying (60/90) ...
or clamd not found yet, retrying (61/90) ...
or clamd not found yet, retrying (62/90) ...
or clamd not found yet, retrying (63/90) ...
or clamd not found yet, retrying (64/90) ...
or clamd not found yet, retrying (65/90) ...
or clamd not found yet, retrying (66/90) ...
or clamd not found yet, retrying (67/90) ...
or clamd not found yet, retrying (68/90) ...
or clamd not found yet, retrying (69/90) ...
or clamd not found yet, retrying (70/90) ...
or clamd not found yet, retrying (71/90) ...
or clamd not found yet, retrying (72/90) ...
or clamd not found yet, retrying (73/90) ...
or clamd not found yet, retrying (74/90) ...
or clamd not found yet, retrying (75/90) ...
or clamd not found yet, retrying (76/90) ...
or clamd not found yet, retrying (77/90) ...
or clamd not found yet, retrying (78/90) ...
or clamd not found yet, retrying (79/90) ...
or clamd not found yet, retrying (80/90) ...
or clamd not found yet, retrying (81/90) ...
or clamd not found yet, retrying (82/90) ...
or clamd not found yet, retrying (83/90) ...
or clamd not found yet, retrying (84/90) ...
or clamd not found yet, retrying (85/90) ...
or clamd not found yet, retrying (86/90) ...
or clamd not found yet, retrying (87/90) ...
or clamd not found yet, retrying (88/90) ...
or clamd not found yet, retrying (89/90) ...
or clamd not found yet, retrying (90/90) ...
Failed to start clamd
Starting Freshclamd
Starting ClamAV

or clamd not found yet, retrying (0/90) ...ClamAV update process started at Fri Mar 29 13:58:52 2024
daily.cld database is up-to-date (version: 27229, sigs: 2057112, f-level: 90, builder: raynman)
main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
bytecode.cld database is up-to-date (version: 335, sigs: 86, f-level: 90, builder: raynman)
LibClamAV Error: Can't load /var/lib/clamav/main.cvd: Can't verify database integrity
LibClamAV Error: cli_loaddbdir: error loading database /var/lib/clamav/main.cvd
ERROR: Fri Mar 29 13:58:59 2024 -> Can't verify database integrity

**Comment by leuedaniel** [comment]
How is your setup in nextcloud? Also like that?:
![image](https://github.com/nextcloud/all-in-one/assets/44936305/0aa23ce8-18d6-402e-be89-c702535f378a)

**Comment by Sabering1** [comment]
Yea, even logs display it
![image](https://github.com/nextcloud/all-in-one/assets/39810634/e5861df5-8de7-478e-83f3-2d11be76a49c)

**Comment by conloos** [comment]
Hi all,

i have the same issue. The setup checked and like the Screenshot.

Full output of the Error Message:

![Screenshot_20240330_112551](https://github.com/nextcloud/all-in-one/assets/524257/b3060a70-b2ba-466d-9616-6d5212a0ff05)

**Comment by ShapeShifter499** [comment]
I'm seeing the same issue here, the clamav container isn't starting for some reason

**Comment by johndoejai** [comment]
https://github.com/nextcloud/files_antivirus/issues/305

**Comment by szaimen** [comment]
Hi, I cannot reproduce this on my test instance. It works correctly without any problem. 

Maybe all of you actually do not have enough RAM or the clamav services were not available so that the database could not be checked?

**Reply by johndoejai** [reply]
In my case there is sufficient RAM available (8 GB).
There also no entries in freshclam.log and clamd.log , which give me a hint.
I only see a clamd-"zombie" after some amount of time:
```sh
$ ps                                                                           
PID   USER     TIME  COMMAND                                                     
    1 clamav    0:03 {init} /sbin/tini /bin/sh /init                             
    6 clamav    0:03 tail -f /dev/null                                           
   10 clamav    0:00 freshclam --checks=1 --daemon --foreground --stdout --user= 
   11 clamav    5:02 [clamd]                                                
20491 clamav    0:00 sh                                                          
20566 clamav    0:00 ps                                                          
```
Any idea, where to look furtther to find the cause?

**Reply by szaimen** [reply]
This sounds like a bug in clamav. Feel free to report this to https://github.com/Cisco-Talos/clamav-docker/issues

**Reply by ShapeShifter499** [reply]
I happen to have 16GB of RAM here with ZRAM and 64GB of backing SWAP on my system

**Reply by ShapeShifter499** [reply]
I take this back, I checked and clamAV has started working again.

```
WARNING: Clamd was NOT notified: Can't connect to clamd through /tmp/clamd.sock: No such file or directory

or clamd not found yet, retrying (18/90) ...
or clamd not found yet, retrying (19/90) ...
or clamd not found yet, retrying (20/90) ...
or clamd not found yet, retrying (21/90) ...
or clamd not found yet, retrying (22/90) ...
or clamd not found yet, retrying (23/90) ...
or clamd not found yet, retrying (24/90) ...
or clamd not found yet, retrying (25/90) ...
or clamd not found yet, retrying (26/90) ...
or clamd not found yet, retrying (27/90) ...
or clamd not found yet, retrying (28/90) ...
or clamd not found yet, retrying (29/90) ...
or clamd not found yet, retrying (30/90) ...
or clamd not found yet, retrying (31/90) ...
or clamd not found yet, retrying (32/90) ...
or clamd not found yet, retrying (33/90) ...
or clamd not found yet, retrying (34/90) ...
or clamd not found yet, retrying (35/90) ...
or clamd not found yet, retrying (36/90) ...
or clamd not found yet, retrying (37/90) ...
or clamd not found yet, retrying (38/90) ...
or clamd not found yet, retrying (39/90) ...
or clamd not found yet, retrying (40/90) ...
or clamd not found yet, retrying (41/90) ...
or clamd not found yet, retrying (42/90) ...
or clamd not found yet, retrying (43/90) ...
or clamd not found yet, retrying (44/90) ...Tue Apr  2 04:00:47 2024 -> Limits: Global time limit set to 120000 milliseconds.
Tue Apr  2 04:00:47 2024 -> Limits: Global size limit set to 419430400 bytes.
Tue Apr  2 04:00:47 2024 -> Limits: File size limit set to 104857600 bytes.
Tue Apr  2 04:00:47 2024 -> Limits: Recursion level limit set to 17.
Tue Apr  2 04:00:47 2024 -> Limits: Files limit set to 10000.
Tue Apr  2 04:00:47 2024 -> Limits: MaxEmbeddedPE limit set to 41943040 bytes.
Tue Apr  2 04:00:47 2024 -> Limits: MaxHTMLNormalize limit set to 41943040 bytes.
Tue Apr  2 04:00:47 2024 -> Limits: MaxHTMLNoTags limit set to 8388608 bytes.
Tue Apr  2 04:00:47 2024 -> Limits: MaxScriptNormalize limit set to 20971520 bytes.
Tue Apr  2 04:00:47 2024 -> Limits: MaxZipTypeRcg limit set to 1048576 bytes.
Tue Apr  2 04:00:47 2024 -> Limits: MaxPartitions limit set to 50.
Tue Apr  2 04:00:47 2024 -> Limits: MaxIconsPE limit set to 100.
Tue Apr  2 04:00:47 2024 -> Limits: MaxRecHWP3 limit set to 16.
Tue Apr  2 04:00:47 2024 -> Limits: PCREMatchLimit limit set to 100000.
Tue Apr  2 04:00:47 2024 -> Limits: PCRERecMatchLimit limit set to 2000.
Tue Apr  2 04:00:47 2024 -> Limits: PCREMaxFileSize limit set to 104857600.
Tue Apr  2 04:00:47 2024 -> Archive support enabled.
Tue Apr  2 04:00:47 2024 -> AlertExceedsMax heuristic detection disabled.
Tue Apr  2 04:00:47 2024 -> Heuristic alerts enabled.
Tue Apr  2 04:00:47 2024 -> Portable Executable support enabled.
Tue Apr  2 04:00:47 2024 -> ELF support enabled.
Tue Apr  2 04:00:47 2024 -> Mail files support enabled.
Tue Apr  2 04:00:47 2024 -> OLE2 support enabled.
Tue Apr  2 04:00:47 2024 -> PDF support enabled.
Tue Apr  2 04:00:47 2024 -> SWF support enabled.
Tue Apr  2 04:00:47 2024 -> HTML support enabled.
Tue Apr  2 04:00:47 2024 -> XMLDOCS support enabled.
Tue Apr  2 04:00:47 2024 -> HWP3 support enabled.
Tue Apr  2 04:00:47 2024 -> OneNote support enabled.
Tue Apr  2 04:00:47 2024 -> Self checking every 600 seconds.
Tue Apr  2 04:00:47 2024 -> Set stacksize to 1048576
socket found, clamd started.
Tue Apr  2 04:11:04 2024 -> SelfCheck: Database status OK.
Tue Apr  2 04:21:05 2024 -> SelfCheck: Database status OK.
Tue Apr  2 04:31:06 2024 -> SelfCheck: Database status OK.
Tue Apr  2 04:41:08 2024 -> SelfCheck: Database status OK.
Tue Apr  2 04:51:09 2024 -> SelfCheck: Database status OK.
Tue Apr  2 05:01:11 2024 -> SelfCheck: Database status OK.
Tue Apr  2 05:11:12 2024 -> SelfCheck: Database status OK.
Tue Apr  2 05:21:14 2024 -> SelfCheck: Database status OK.
Tue Apr  2 05:31:15 2024 -> SelfCheck: Database status OK.
Tue Apr  2 05:41:17 2024 -> SelfCheck: Database status OK.
Tue Apr  2 05:51:18 2024 -> SelfCheck: Database status OK.
Tue Apr  2 06:01:20 2024 -> SelfCheck: Database status OK.
Tue Apr  2 06:11:22 2024 -> SelfCheck: Database status OK.
Tue Apr  2 06:21:24 2024 -> SelfCheck: Database status OK.
Tue Apr  2 06:31:26 2024 -> SelfCheck: Database status OK.
Tue Apr  2 06:41:28 2024 -> SelfCheck: Database status OK.
Tue Apr  2 06:51:31 2024 -> SelfCheck: Database status OK.
Tue Apr  2 07:01:33 2024 -> SelfCheck: Database status OK.
Tue Apr  2 07:11:35 2024 -> SelfCheck: Database status OK.
Tue Apr  2 07:21:36 2024 -> SelfCheck: Database status OK.
Tue Apr  2 07:31:38 2024 -> SelfCheck: Database status OK.
Tue Apr  2 07:41:39 2024 -> SelfCheck: Database status OK.
Tue Apr  2 07:51:41 2024 -> SelfCheck: Database status OK.
Tue Apr  2 08:01:42 2024 -> SelfCheck: Database status OK.
Tue Apr  2 08:11:44 2024 -> SelfCheck: Database status OK.
Tue Apr  2 08:21:45 2024 -> SelfCheck: Database status OK.
Tue Apr  2 08:31:47 2024 -> SelfCheck: Database status OK.
Tue Apr  2 08:41:49 2024 -> SelfCheck: Database status OK.
Tue Apr  2 08:51:50 2024 -> SelfCheck: Database status OK.
Tue Apr  2 09:01:52 2024 -> SelfCheck: Database status OK.
Tue Apr  2 09:11:53 2024 -> SelfCheck: Database status OK.
Tue Apr  2 09:21:55 2024 -> SelfCheck: Database status OK.
Tue Apr  2 09:31:57 2024 -> SelfCheck: Database status OK.
Tue Apr  2 09:41:58 2024 -> SelfCheck: Database status OK.
Tue Apr  2 09:52:00 2024 -> SelfCheck: Database status OK.
Tue Apr  2 10:02:01 2024 -> SelfCheck: Database status OK.
Tue Apr  2 10:12:03 2024 -> SelfCheck: Database status OK.
Tue Apr  2 10:22:05 2024 -> SelfCheck: Database status OK.
Tue Apr  2 10:32:06 2024 -> SelfCheck: Database status OK.
Tue Apr  2 10:42:08 2024 -> SelfCheck: Database status OK.
Tue Apr  2 10:52:09 2024 -> SelfCheck: Database status OK.
Tue Apr  2 11:02:11 2024 -> SelfCheck: Database status OK.
Tue Apr  2 11:12:13 2024 -> SelfCheck: Database status OK.
Tue Apr  2 11:22:14 2024 -> SelfCheck: Database status OK.
Tue Apr  2 11:32:16 2024 -> SelfCheck: Database status OK.
Tue Apr  2 11:42:18 2024 -> SelfCheck: Database status OK.
Tue Apr  2 11:52:19 2024 -> SelfCheck: Database status OK.
Tue Apr  2 12:02:21 2024 -> SelfCheck: Database status OK.
Tue Apr  2 12:12:23 2024 -> SelfCheck: Database status OK.
Tue Apr  2 12:22:25 2024 -> SelfCheck: Database status OK.
Tue Apr  2 12:32:27 2024 -> SelfCheck: Database status OK.
Tue Apr  2 12:42:29 2024 -> SelfCheck: Database status OK.
Tue Apr  2 12:52:31 2024 -> SelfCheck: Database status OK.
Tue Apr  2 13:02:32 2024 -> SelfCheck: Database status OK.
Tue Apr  2 13:12:34 2024 -> SelfCheck: Database status OK.
Tue Apr  2 13:22:36 2024 -> SelfCheck: Database status OK.
Tue Apr  2 13:32:38 2024 -> SelfCheck: Database status OK.
Tue Apr  2 13:42:40 2024 -> SelfCheck: Database status OK.
Tue Apr  2 13:52:42 2024 -> SelfCheck: Database status OK.
Tue Apr  2 14:02:44 2024 -> SelfCheck: Database status OK.
Tue Apr  2 14:12:46 2024 -> SelfCheck: Database status OK.
Tue Apr  2 14:22:47 2024 -> SelfCheck: Database status OK.
Tue Apr  2 14:32:49 2024 -> SelfCheck: Database status OK.
Tue Apr  2 14:42:51 2024 -> SelfCheck: Database status OK.
Tue Apr  2 14:52:53 2024 -> SelfCheck: Database status OK.
Tue Apr  2 15:02:55 2024 -> SelfCheck: Database status OK.
Tue Apr  2 15:12:57 2024 -> SelfCheck: Database status OK.
Tue Apr  2 15:22:59 2024 -> SelfCheck: Database status OK.
Tue Apr  2 15:33:01 2024 -> SelfCheck: Database status OK.
Tue Apr  2 15:43:04 2024 -> SelfCheck: Database status OK.
Tue Apr  2 15:53:05 2024 -> SelfCheck: Database status OK.
Tue Apr  2 16:03:08 2024 -> SelfCheck: Database status OK.
Tue Apr  2 16:13:09 2024 -> SelfCheck: Database status OK.
Tue Apr  2 16:23:12 2024 -> SelfCheck: Database status OK.
Tue Apr  2 16:33:14 2024 -> SelfCheck: Database status OK.
```

**Comment by ShapeShifter499** [comment]
> Maybe all of you actually do not have enough RAM or the clamav services were not available so that the database could not be checked?

Is this something that could happen if it couldn't download the virus database?   I have a daily update setup on my system, maybe the services went down right as the ClamAV container along with everything else got redone when the update script was called. If not I'm not sure what else caused it.  It is now working fine again.

**Comment by MyNameIsRatchet** [comment]
I came from [here](https://github.com/nextcloud/files_antivirus/issues/305#issuecomment-2028583630), we have this problem since 2 weeks now and I ended up disabling the background scan. I assume the failure started with the latest files_antivirus-App for nextcloud, since AIO is also using this app the developers could maybe look into this?

**Reply by leuedaniel** [reply]
Since aio Update tomorrow everything works fine.

**Comment by leuedaniel** [comment]
Sorry I have to correct myself again. After starting the containers, the ClamAV container was healthy and in Nextcloud I was able to save the connection to the ClamAV container and everything was ok. After a certain time, however, the ClamAV container is unhealthy and when saving the connection to the container in Nextcloud the message appears again: "The ClamAV module is not in daemon mode."

**Comment by mhl66** [comment]
Same for me: first it worked, then it went to unhealthy state, and the logs are exploding (see above)...

**Comment by szaimen** [comment, marked as answer]
This seems to be an upstream issue: https://github.com/Cisco-Talos/clamav-docker/issues/45

> after switching to clamav 1.3, we have seen "disappearing" clamds -- for a reason yet to be researched.
